src/handlers/mod.rs

```diff
@@ -425,3 +428,56 @@ impl DrmLeaseHandler for State {
     }
 }
 delegate_drm_lease!(State);
+
+impl GammaControlHandler for State {
+    fn gamma_control_manager_state(&mut self) -> &mut GammaControlManagerState {
+        &mut self.niri.gamma_control_manager_state
+    }
+
+    fn set_gamma(&mut self, wl_output: &WlOutput, ramp: Vec<u16>) -> Result<(), ()> {
+        if !self.backend.is_tty() {
```

Comment:
Instead of these checks here, just make it so the global is only available when using the tty backend.

Reply:
How would I do that, can you point me to a place where that has been done?

Reply:
Several ways to do it. For example, can store an Option<GammaControlState> on Niri, and only create it if it's tty. Or add a global filter (should be done anyway for restricted clients, same as in foreign toplevel management) and always return false if backend is not tty.

Reply:
Sorry, still not getting it...

What exactly are you referring to, when saying global? The GlobalDispatch? If that is the case, changing the delegate_gamma_control conditionally seems impossible to me, as that is happening before we have a state.

What is a global filter?

Reply:
This is what I'm referring to:

https://github.com/YaLTeR/niri/blob/1a784e6e66785f360da6df59ae2fb4b98370ca3c/src/niri.rs#L909-L912

This closure controls which clients can see the global. In this case, restricted clients (i.e. sandboxed with flatpak) cannot see the global. I suggest adding that and additionally always returning false from that closure if the backend is not TTY, this way on winit clients will simply not see it.

Implementing that closure is very simple:
1. https://github.com/YaLTeR/niri/blob/1a784e6e66785f360da6df59ae2fb4b98370ca3c/src/protocols/foreign_toplevel.rs#L63-L66
2. https://github.com/YaLTeR/niri/blob/1a784e6e66785f360da6df59ae2fb4b98370ca3c/src/protocols/foreign_toplevel.rs#L327-L329

Reply:
This is what I'm referring to:

https://github.com/YaLTeR/niri/blob/1a784e6e66785f360da6df59ae2fb4b98370ca3c/src/niri.rs#L909-L912

This closure controls which clients can see the global. In this case, restricted clients (i.e. sandboxed with flatpak) cannot see the global. I suggest adding that and additionally always returning false from that closure if the backend is not TTY, this way on winit clients will simply not see it.

Implementing that closure is very simple:
1. https://github.com/YaLTeR/niri/blob/1a784e6e66785f360da6df59ae2fb4b98370ca3c/src/protocols/foreign_toplevel.rs#L63-L66
2. https://github.com/YaLTeR/niri/blob/1a784e6e66785f360da6df59ae2fb4b98370ca3c/src/protocols/foreign_toplevel.rs#L327-L329